pkg/pulumiyaml/run.go

```diff
@@ -685,6 +685,37 @@ func (r *Runner) Evaluate(ctx *pulumi.Context) syntax.Diagnostics {
 	return r.Run(programEvaluator{evalContext: eCtx, pulumiCtx: ctx})
 }
 
+func (r *Runner) getPulumiConfNodes() []configNode {
+	var projConf map[string]interface{}
+	var projConfStr = os.Getenv("PULUMI_CONFIG")
+	nodes := make([]configNode, 0)
```

Comment:
If were going to use the `make([]type, length)` syntax here, we can use it to ensure a single allocation by moving it after we deserialize `projConfStr`:
```suggestion
	nodes := make([]configNode, len(projConf))
```